code/src/components/Moviedetail.js

```diff
@@ -0,0 +1,194 @@
+import React, { useEffect, useState } from 'react'
+import { Link, useParams } from 'react-router-dom'
+import styled from 'styled-components'
+
+import { API_KEY } from 'Apis/Urls';
+
+const Moviedetail = () => {
+    const { movieId } = useParams();
+    
+    const [movieDetails, setMovieDetails] = useState([]);
+
+    useEffect(() => {
+        fetch(`https://api.themoviedb.org/3/movie/${movieId}?api_key=${API_KEY}&language=en-US`)
+        .then(res => res.json())
+        .then(movieInfo => {
+            setMovieDetails(movieInfo)
+        })
+    }, [movieId])
+
+    const StyledBackLink = styled(Link)`
+        position: absolute;        
+        width: 200px;
+        margin: 30px;      
+        color: white;
+        font-weight: 700;  
+        text-decoration: none;
+        border-radius: 10px;
+        text-align: center;
+        padding: 10px;
+        display: flex;        
+
+        &:hover {      
+            transition: 0.5s ease;      
+            filter: invert(70%);
+        }
+
+        @media (min-width: 768px) {
+            font-size: 30px;
+        }
+    ` 
+    
+    const Div = styled.div`
+        position:fixed;
+        margin: 0;
+        display: flex;
+        flex-direction: column;
+        color: white;
+        height: 100vh;
+        width: 100vw;
+        
+        
+        
+    `
+
+    const BackdropImage = styled.img`               
+        height: 100vh;        
+        object-fit: cover;
+    `
+
+    const MovieInfo = styled.section`
+        position: absolute;
+        margin-left: 30px;
+        display: flex;
+        flex-direction: column;
+        justify-content: flex-end;
+        height: 92vh;
+        
+        @media (min-width: 550px) {
+            margin-left: 30px;               
+        }
+
+        @media (min-width: 768px) {
+            margin-left: 30px; 
+            display: flex;
+            flex-direction: column;
+            justify-content: flex-end;     
+        }
+    `
+    
+    const MovieSmallImage = styled.img`
+        position: absolute;
+        border: 3px solid white;
+        border-radius: 10px;
+        width: 200px;
+
+        @media (max-width: 768px) {
+            width: 150px;
+            margin-right: 0;
+        }
+
+        @media (min-width: 375px) and (max-width: 767px) {
+            display: none;
+        }
+    `
+
+    const TextContainer = styled.section`
+        position: absolute;
+        display: flex;
+        flex-direction: column;
+        width: 80vw;
+        background-color: rgba(0, 0, 0, 0.637);
+        border-radius: 10px;          
+
+        @media (min-width: 768px) {
+            margin-left: 30vw;
+            width: 55vw;
```

Comment:
When I tested your site on desktop view, it works perfectly. However, I would take a closer look into mobile view especially on Safari. I tested the site on iPhone12 Pro and the movie rating seems to be cut in half.